source/contracts/trading/CancelOrder.sol

```diff
@@ -21,12 +22,12 @@ import 'libraries/Extractable.sol';
  * @title CancelOrder
  * @dev This allows you to cancel orders on the book.
  */
-contract CancelOrder is CashAutoConverter, Extractable, ReentrancyGuard, ICancelOrder {
+contract CancelOrder is CashAutoConverter, Extractable, ReentrancyGuard, MarketValidator, ICancelOrder {
     /**
      * @dev Cancellation: cancels an order, if a bid refunds money, if an ask returns shares
      * @return true if successful; throw on failure
      */
-    function cancelOrder(bytes32 _orderId, Order.Types _type, IMarket _market, uint8 _outcome) nonReentrant convertToAndFromCash external returns (bool) {
+    function cancelOrder(bytes32 _orderId, Order.Types _type, IMarket _market, uint8 _outcome) marketIsValid(_market) nonReentrant convertToAndFromCash external returns (bool) {
```

Comment:
Did you already do a find-in-files for functions that take an `IMarket` parameter?

I wonder if we should also have a similar modifier for `IUniverse` and apply it everywhere?

Reply:
Ill double check!